services/beaconchain/src/main/java/tech/pegasys/teku/services/beaconchain/BeaconChainController.java

```diff
@@ -642,6 +643,13 @@ protected void initAttesterSlashingPool() {
                 .reversed());
     blockImporter.subscribeToVerifiedBlockAttesterSlashings(attesterSlashingPool::removeAll);
     attesterSlashingPool.subscribeOperationAdded(forkChoice::onAttesterSlashing);
+    if (beaconConfig.validatorConfig().isShutdownWhenValidatorSlashedEnabled()) {
+      final ValidatorTimingChannel validatorTimingChannel =
+          eventChannels.getPublisher(ValidatorTimingChannel.class);
+      attesterSlashingPool.subscribeOperationAdded(
+          (operation, validationStatus, fromNetwork) ->
+              validatorTimingChannel.onAttesterSlashing(operation));
+    }
```

Comment:
iirc this is where we'd need a local AT to show these (at least proposer) work... not sure how good our solution currently is for unit testing `BeaconChainController`
Arguably a bit of a gap in our infra but outside the scope of your update. I guess the problem ultimately is it would be very easy for this to disappear and us break the feature without knowing, and BN+VC is the default mode of operation.

Reply:
It would be good to make `BeaconChainController` testable one day

Reply:
I agree, historically we break out things that need testing into their own class but there's instance like this where it'd be neat.